torch/nn/modules/transformer.py

```diff
@@ -267,6 +267,24 @@ class TransformerEncoderLayer(Module):
     Neural Information Processing Systems, pages 6000-6010. Users may modify or implement
     in a different way during application.
 
+    forward() will use a special optimized implementation if all of the following
+    conditions are met:
+    - inference mode is enabled (using the torch.inference_mode() context manager)
```

Comment:
The code fragments here will format better if they are surrounded by double backticks

Reply:
I think I mentioned this when we were chatting, but to record it here: I think it's a little surprising that inference mode has to be enabled to hit the optimized implementation; just testing that the inputs don't require grad should be enough.

Reply:
> just testing that the inputs don't require grad should be enough

don't we want people to use torch.inference_mode, not torch.no_grad? this seems like a good way to push them toward the right behavior.